src/axom/multimat/multimat.cpp

```diff
@@ -382,16 +460,53 @@ void MultiMat::setCellMatRel(const vector<bool>& vecarr, DataLayout layout)
     counts[i] = cardinality;
   }
 
+  if(AllocatorOnDevice(m_slamAllocatorId))
+  {
+    // Copy counts and indices to GPU memory.
+    counts = IndBufferType(counts, m_slamAllocatorId);
+    indices = IndBufferType(indices, m_slamAllocatorId);
+  }
+
   setCellMatRel(counts, indices, layout);
 }
 
+template <typename ExecSpace, typename IndexType>
```

Comment:
Please add a brief function comment (even though this won't get generated in the doxygen).